src/riak_kv_index_hashtree.erl

```diff
@@ -1029,3 +1039,8 @@ maybe_get_vnode_lock(SrcPartition, Pid) ->
         false ->
             ok
     end.
+
+maybe_callback(undefined) ->
+    ok;
+maybe_callback(Callback) ->
+    Callback().
```

Comment:
This should probably be `catch Callback().` unless we're expecting the consumer to be able to cancel the update by throwing (which would seem more appropriate to do with a return value from Callback() anyway).